src/Hosting/Hosting/test/HostTests.cs

```diff
@@ -85,21 +85,23 @@ public void CreateDefaultBuilder_EnablesValidateOnBuild()
         [Fact]
         public void CreateDefaultBuilder_ConfigJsonDoesNotReload()
         {
-            Environment.SetEnvironmentVariable("ASPNETCORE_BUILDER_CONFIG_RELOAD", "false");
+            Environment.SetEnvironmentVariable("DOTNET_HOSTBUILDER_CONFIG_RELOAD", "false");
```

Comment:
Tests shouldn't alter environment variables, it can have side-effects on other tests. Since we're using the config system now provide the key via AddInMemoryCollection.
https://github.com/dotnet/extensions/blob/126381c5526d59eb573386ffc642820091555c32/src/Hosting/Hosting/test/HostBuilderTests.cs#L39-L45

Reply:
I will update the config to polyfill from a config source like an in memory collection instead as you are suggesting, but this will then drop the assertion that DOTNET_HOSTBUILDER_CONFIG_RELOAD works, and moves it to being untested, and unverifiable. I'm not sure of a good solution here as this means that any tests that would assert that `config.AddEnvironmentVariables(prefix: "DOTNET_");` works in `CreateDefaultBuilder()`, or anywhere else are untestable :(